test/e2e/storage/testsuites/snapshotsecrets.go

```diff
@@ -0,0 +1,253 @@
+/*
+Copyright 2018 The Kubernetes Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package testsuites
+
+import (
+	"context"
+
+	"time"
+
+	"github.com/onsi/ginkgo"
+	v1 "k8s.io/api/core/v1"
+	"k8s.io/client-go/dynamic"
+
+	apierrors "k8s.io/apimachinery/pkg/api/errors"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+
+	"k8s.io/apimachinery/pkg/apis/meta/v1/unstructured"
+
+	clientset "k8s.io/client-go/kubernetes"
+	"k8s.io/kubernetes/test/e2e/framework"
+
+	e2epv "k8s.io/kubernetes/test/e2e/framework/pv"
+	e2eskipper "k8s.io/kubernetes/test/e2e/framework/skipper"
+	e2evolume "k8s.io/kubernetes/test/e2e/framework/volume"
+	"k8s.io/kubernetes/test/e2e/storage/testpatterns"
+	"k8s.io/kubernetes/test/e2e/storage/utils"
+)
+
+type snapshotsecretsTestSuite struct {
+	tsInfo TestSuiteInfo
+}
+
+var _ TestSuite = &snapshotsecretsTestSuite{}
+
+// InitSnapshotsecretsTestSuite returns snapshottableTestSuite that implements TestSuite interface
+func InitSnapshotsecretsTestSuite() TestSuite {
+	return &snapshotsecretsTestSuite{
+		tsInfo: TestSuiteInfo{
+			Name: "snapshotsecrets",
+			TestPatterns: []testpatterns.TestPattern{
+				// testpatterns.DynamicSnapshotDelete,
+				testpatterns.DynamicSnapshotRetain,
+				// testpatterns.PreprovisionedSnapshotDelete,
+				// testpatterns.PreprovisionedSnapshotRetain,
```

Comment:
We should also have tests for DynamicSnapshotDelete, PreprovisionedSnapshotDelete, and PreprovisionedSnapshotRetain.  In the pre-provisioned cases, the secrets annotation needs to be applied manually by the user.